pytorch_lightning/callbacks/progress.py

```diff
@@ -7,7 +7,14 @@
 """
 import sys
 
-from tqdm.auto import tqdm
+try:
+    # try importing ipywidgets before importing tqdm
+    # to ensure it won't fail and a progress bar is displayed
+    import ipywidgets
+    from tqdm.auto import tqdm
+except ImportError as e:
```

Comment:
could we use 
```python
import importlib 
if importlib.util.find_spec("ipywidgets") is not None:
    from tqdm.auto import tqdm
else:
    from tqdm import tqdm
```
? (brain compiled, may have typos)
ImportError is not suitable here in general.

Reply:
should probably also add "pragma no cover " not to decrease coverage

Reply:
Yes, sure. Thanks!